core-java-modules/core-java-date-operations-4/src/test/java/com/baeldung/timestamptocalendar/SqlTimestampToCalendarConverterUnitTest.java

```diff
@@ -0,0 +1,37 @@
+package com.baeldung.timestamptocalendar;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+import java.sql.Timestamp;
+import java.util.Calendar;
+
+import org.junit.Test;
+
+public class SqlTimestampToCalendarConverterUnitTest {
+
+    @Test
+    public void givenTimestamp_whenConvert_thenEqualMillis() {
+        Timestamp timestamp = new Timestamp(124, 3, 19, 9, 30, 0, 801789562);
+        Calendar calendar = SqlTimestampToCalendarConverter.timestampToCalendar(timestamp);
+        assertEquals(calendar.getTimeInMillis(), timestamp.getTime());
+    }
+
+    @Test
+    public void givenCalendarFromTimestamp_whenConvertBack_thenEqualMillis() {
+        Timestamp timestamp = new Timestamp(124, 3, 19, 9, 30, 0, 801789562);
+        Calendar calendar = SqlTimestampToCalendarConverter.timestampToCalendar(timestamp);
+        timestamp = SqlTimestampToCalendarConverter.calendarToTimestamp(calendar);
+        assertEquals(calendar.getTimeInMillis(), timestamp.getTime());
+    }
+
+    @Test
+    public void givenCalendar_whenConvert_thenLoseNanos() {
+        int nanos = 801789562;
+        Timestamp timestamp = new Timestamp(124, 3, 19, 9, 30, 0, nanos);
+        assertEquals(nanos, timestamp.getNanos());
+        Calendar calendar = SqlTimestampToCalendarConverter.timestampToCalendar(timestamp);
+        timestamp = SqlTimestampToCalendarConverter.calendarToTimestamp(calendar);
+        assertNotEquals(nanos, timestamp.getNanos());
```

Comment:
Let's do an assertEquals here instead to make it clear what exactly happens to the nanos field vs the original value